### PR TITLE
Add Testcase to verify JSON support in NashornJs

### PR DESCRIPTION
**Pull request by malakaganga**

### CODE REVIEW

File: integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/script/NativeJsonSupportByNashornJsTestCase.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            LoginAuthenticationExceptionException, XMLStreamException, AutomationFrameworkException {
+
+        String payload = "{\n" + "\"name\": \"John Doe\",\n" + "\"dob\": \"1990-03-19\",\n" + "\"ssn\": "
+                + "\"234-23" + "-525\",\n" + "\"address\": \"California\",\n" + "\"phone\": \"8770586755\",\n"
+                + "\"email\":" + " \"johndoe@gmail.com\",\n" + "\"doctor\": \"thomas collins\",\n" + "\"hospital\": "
+                + "\"grand oak " + "community hospital\",\n" + "\"cardNo\": \"7844481124110331\",\n" +
+                "\"appointment_date\": " + "\"2017-04-02\"\n" + "}";
+        HttpResponse response = doPost(new URL(getApiInvocationURL("nashornJsNativeJSONSupport")), payload,
+                httpHeaders);
+        Assert.assertTrue((response.getData().contains("California")), "Validation must be success. Response: "
```
Comment (Subasinghe) [inline integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/script/NativeJsonSupportByNashornJsTestCase.java]: Shall we add a meaningful failure string here with related to comparing string "California" ?

File: integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/script/NativeJsonSupportByNashornJsTestCase.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import java.util.Map;
+import static org.wso2.carbon.automation.test.utils.http.client.HttpRequestUtil.doPost;
+
+/**
+ * This test case verifies that, native json operations are supported by NashornJs.
+ */
+public class NativeJsonSupportByNashornJsTestCase extends ESBIntegrationTest {
+
+    private Map<String, String> httpHeaders = new HashMap<String, String>();
+
```
Comment (Manuri) [inline integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/script/NativeJsonSupportByNashornJsTestCase.java]: Could use <> notation here

File: integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/script/NativeJsonSupportByNashornJsTestCase.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    @BeforeClass(alwaysRun = true)
+    protected void init() throws Exception {
+        super.init();
+        httpHeaders.put("Content-Type", "application/json");
```
Comment (sdkottegoda) [inline integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/script/NativeJsonSupportByNashornJsTestCase.java]: Why do we need to add the headers in the init method?

File: integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/script/NativeJsonSupportByNashornJsTestCase.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    public void testSendingPayloadJson() throws Exception {
+
+        String payload = "{\n" + "\"name\": \"John Doe\",\n" + "\"dob\": \"1990-03-19\",\n" + "\"ssn\": "
+                + "\"234-23" + "-525\",\n" + "\"address\": \"California\",\n" + "\"phone\": \"8770586755\",\n"
+                + "\"email\":" + " \"johndoe@gmail.com\",\n" + "\"doctor\": \"thomas collins\",\n" + "\"hospital\": "
+                + "\"grand oak " + "community hospital\",\n" + "\"cardNo\": \"7844481124110331\",\n"
+                + "\"appointment_date\": " + "\"2017-04-02\"\n" + "}";
+        HttpResponse response = doPost(new URL(getApiInvocationURL("nashornJsNativeJSONSupport")), payload,
+                httpHeaders);
+        Assert.assertTrue((response.getData().contains("California")), "Fails if the response does not contain "
```
Comment (sdkottegoda) [inline integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/script/NativeJsonSupportByNashornJsTestCase.java]: The message should something that gets printed when the assertion fails. e.g. "Response does not contain the keyword: California"

**Pull request merged**